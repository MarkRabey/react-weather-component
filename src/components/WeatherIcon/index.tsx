import React from 'react';
import {Image} from 'react-native';

interface WeatherIconProps {
  /**
   * Icon to display
   * */
  icon: string;
  description: string;
}

const WeatherIcon: React.FC<WeatherIconProps> = ({icon, description}) => (
  <Image
    source={{
      uri: `https://openweathermap.org/img/wn/${icon}@2x.png`,
    }}
    accessibilityLabel={description}
    style={{width: 50, height: 50}}
  />
);

export default WeatherIcon;
