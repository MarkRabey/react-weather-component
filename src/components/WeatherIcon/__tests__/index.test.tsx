import React from 'react';
import renderer from 'react-test-renderer';

import WeatherIcon from '..';

describe('WeatherIcon', () => {
  it('renders correctly', () => {
    const result = renderer
      .create(<WeatherIcon icon="10" description="icon" />)
      .toJSON();
    console.log({result});

    expect(result).toMatchSnapshot();
  });
});
