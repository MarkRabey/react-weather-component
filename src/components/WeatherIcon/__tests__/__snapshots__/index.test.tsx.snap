// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherIcon renders correctly 1`] = `
<div
  aria-label="icon"
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitFlexBasis": "auto",
      "flexBasis": "auto",
      "height": "50px",
      "msFlexPreferredSize": "auto",
      "overflowX": "hidden",
      "overflowY": "hidden",
      "width": "50px",
      "zIndex": 0,
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.00)",
        "backgroundPosition": "center",
        "backgroundRepeat": "no-repeat",
        "backgroundSize": "cover",
        "bottom": "0px",
        "height": "100%",
        "left": "0px",
        "position": "absolute",
        "right": "0px",
        "top": "0px",
        "width": "100%",
        "zIndex": -1,
      }
    }
    suppressHydrationWarning={true}
  />
</div>
`;
